scrapi/harvesters/lwbin.py

```diff
@@ -0,0 +1,210 @@
+"""
+A Lake Winnipeg Basin Information Network (BIN) harvester for the SHARE project
+
+Example API request: http://130.179.67.140/api/3/action/package_search?q= (problematic)
+http://130.179.67.140/api/3/action/current_package_list_with_resources (currently using)
+It oddly returns 5 more datasets than all searchable ones on LWBIN data hub.
+
+Known issues:
+1 -- Five datasets can be searched but cannot be accessed via LWBIN.
+Clicking on the searching result would result in linking to a redirected page like this:
+http://130.179.67.140/user/login?came_from=http://130.179.67.140/dataset/mpca-surface-water-data-access-interactive-map
+Within each dataset there are resouces that contain urls to source pages. For future work considering using resources
+urls as canonical urls.
+2 -- Resouces properties contained in raw metadata of the datasets are not added to the normalized metadata at this
+point.
+3 -- Single name contributors can be used as filters or an invalid query will be returned. Has nothing to do with scrapi but the frontend.
+"""
+
+from __future__ import unicode_literals
+
+import json
+import logging
+
+from nameparser import HumanName
+from dateutil.parser import parse
+
+from scrapi import requests
+from scrapi.base import JSONHarvester
+from scrapi.linter.document import RawDocument
+from scrapi.base.helpers import build_properties, date_formatter
+
+
+logger = logging.getLogger(__name__)
+
+ORGANIZATIONS = (
+    "organization", "Organization", "fund", "Fund", "canada", "Canada", "agriculture", "Agriculture", "commitee",
+    "Commitee", "international", "International", "council", "Council", "office", "Office", "of", "Of",
+    "observation", "Observation", "LWBIN", "CoCoRaHS", "USGS", "NSIDC"
+)
+
+
+def is_organization(name):
+    """Return a boolean to indicate if the name passed to the function is an organization
+    """
+    words = name.split(' ')
+    return any(word.strip(";") in ORGANIZATIONS for word in words)
+
+
+def process_contributors(authors, emails):
+    """Process authors and add author emails
+    If multiple authors and one email, put email in a new author
+    """
+
+    authors = authors.strip().replace('<span class="author-names">', '').replace('</span>', '')
+    authors = authors.split(',')
+    emails = emails.split(',')
+
+    for author in authors:
+        if is_organization(author):
+            break
+    else:
+        if ' and ' in authors[-1] or ' <em>et al.</em>' in authors[-1]:
+            split_name = authors.pop(-1).replace(' <em>et al.</em>', '').split(' and ')
+            authors.extend(split_name)
+
+    contributor_list = []
+    append_emails = True
+    if len(authors) != 1 or len(emails) != 1 or emails[0] == u'':
+        append_emails = False  # append the email to the author only when 1 record is observed
```

Comment:
you can just say `append_emails = len(authors) != 1 or len(emails) != 1 or not emails`